manager/integration/tests/test_node.py

```diff
@@ -2388,93 +2401,97 @@ def test_node_eviction(client, core_api, volume_name, pod, settings_reset): # NO
               replica_node_soft_anti_affinity_setting)
         print(e)
 
+    # replica now all running on node 2, disable node 2 schedule to
+    # relocate to node 1, 3
     client.update(node2, allowScheduling=False, evictionRequested=True)
 
-    volume1 = client.by_id_volume(volume1_name)
-    assert volume1.robustness == "healthy"
-
-    wait_for_volume_replica_count(client, volume1_name, 3)
-    wait_for_volume_replica_count(client, volume1_name, 2)
-
-    volume1 = client.by_id_volume(volume1_name)
-    assert volume1.robustness == "healthy"
-
-    wait_for_volume_replica_count(client, volume1_name, 3)
+    # relocation from the only replica node of anti-affinity could
+    # spawn single or mutiple replica, hense we do not know the
+    # expected replica size during replication
+    wait_for_volume_replica_count_not_equal(client, volume1_name, 2)
     wait_for_volume_replica_count(client, volume1_name, 2)
 
-    volume1 = client.by_id_volume(volume1_name)
-    assert volume1.robustness == "healthy"
+    # k8s may or may not run an extra relocation
+    try:
+        wait_for_volume_replica_count_not_equal(client, volume1_name, 2)
+        wait_for_volume_replica_count(client, volume1_name, 2)
+    except AssertionError as e:
+        print("\nException when asserting replica count, "
+              "should already correctly allocated",
+              volume1_name)
+        print(e)
+        pass
 
+    volume1 = wait_for_volume_healthy(client, volume1_name)
```

Comment:
Volume should be always healthy during the process?

Reply:
Addressed.